src/optimagic/visualization/backends.py

```diff
@@ -0,0 +1,173 @@
+from typing import Any, Literal, Protocol, runtime_checkable
+
+import plotly.graph_objects as go
+
+from optimagic.config import IS_MATPLOTLIB_INSTALLED
+from optimagic.exceptions import InvalidPlottingBackendError, NotInstalledError
+from optimagic.visualization.plotting_utilities import LineData
+
+if IS_MATPLOTLIB_INSTALLED:
+    import matplotlib as mpl
+    import matplotlib.pyplot as plt
+
+    # Handle the case where matplotlib is used in notebooks (inline backend)
+    # to ensure that interactive mode is disabled to avoid double plotting.
+    # (See: https://github.com/matplotlib/matplotlib/issues/26221)
+    if mpl.get_backend() == "module://matplotlib_inline.backend_inline":
+        plt.install_repl_displayhook()
+        plt.ioff()
```

Comment:
As discussed, we will check how other libraries solve this problem before continuing with this approach.